pkg/kmsg/writer/src/gpud_kmsg_writer.c

```diff
@@ -0,0 +1,141 @@
+#include <linux/fs.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/spinlock.h>
+#include <linux/string.h>
+#include <linux/uaccess.h>
+
+#define DEVICE_NAME "gpud-kmsg-writer"
+#define BUF_LEN 1000
+
+static int major;
+static char msg[BUF_LEN];
+static DEFINE_MUTEX(msg_mutex);
+
+// Function to parse log level from input and return the appropriate KERN_*
+// level.
+// Both of the following commands are valid:
+// e.g.,
+// sh -c "echo \"KERN_EMERG, System critical error!\" > /dev/gpud_kmsg_writer"
+// sh -c "echo \"kern.emerg, System critical error!\" > /dev/gpud_kmsg_writer"
+static const char *parse_log_level(const char *input, char *remaining_msg) {
```

Comment:
Do we really need this kernel module? shall we just write the error message to `/dev/kmsg`?

Reply:
Write some texts to `/dev/kmsg` is doable https://github.com/siderolabs/go-kmsg/blob/main/writer.go, but to simulate like GPU device -> kmsg behavior, we need `printk` function https://github.com/torvalds/linux/blob/master/kernel/printk/printk.c, that supports log level

Reply:
Closing in favor of https://github.com/leptonai/gpud/pull/870.